presto-main/src/main/java/io/prestosql/operator/exchange/LocalExchangeMemoryManager.java

```diff
@@ -16,28 +16,26 @@
 import com.google.common.util.concurrent.ListenableFuture;
 import com.google.common.util.concurrent.SettableFuture;
 
+import javax.annotation.Nullable;
 import javax.annotation.concurrent.GuardedBy;
 import javax.annotation.concurrent.ThreadSafe;
 
 import java.util.concurrent.atomic.AtomicLong;
 
 import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.util.concurrent.Futures.immediateFuture;
 
 @ThreadSafe
 public class LocalExchangeMemoryManager
 {
-    private static final SettableFuture<?> NOT_FULL;
-
-    static {
-        NOT_FULL = SettableFuture.create();
-        NOT_FULL.set(null);
-    }
+    private static final ListenableFuture<?> NOT_BLOCKED = immediateFuture(null);
 
     private final long maxBufferedBytes;
     private final AtomicLong bufferedBytes = new AtomicLong();
 
+    @Nullable
     @GuardedBy("this")
-    private SettableFuture<?> notFullFuture = NOT_FULL;
+    private SettableFuture<?> notFullFuture; // null represents "no callback registered"
```

Comment:
please initialize it with `NOT_BLOCKED`

Reply:
`NOT_BLOCKED` is not a `SettableFuture<?>` but rather a `ListenableFuture<?>` which can't be set (it's already completed). See https://github.com/prestosql/presto/pull/6260#discussion_r547368980 for some additional detail around this implementation strategy.